### PR TITLE
Use Compose on BrowseSourceScreens

### LINKED ISSUE #7418: Adding a manga to library from Global Search / Browse doesn't instantly update it to show "In Library"
### Steps to reproduce

1. Global Search (or Search in a source works too) for anything from any source. Eg: Search Ayashimon (from MangaDex)
2. Open the manga and add it to library
3. Press Back to reach Global Search. Notice that there is no "In Library" blurring shown for the manga you just added to the library
4. Go back to library to see if manga was actually added (spoiler: it is)
5. Now go Global Search for the manga again

### Expected behavior

Manga added to library should show up with a shadowed cover and "in library" badge
This should show up instantly

### Actual behavior

During Step 3, Manga doesn't have "in library" button
After Step 5, it does show up with the "in library" badge

### Crash logs

_No response_

### Tachiyomi version

r4524

### Android version

Android 12

### Device

jasmine_sprout

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.4](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by ghostbear**
Yes it's a big PR but at the same time it affects three screens

closes #7418

### CODE REVIEW

File: app/src/main/java/eu/kanade/presentation/browse/BrowseLatestScreen.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    val context = LocalContext.current
+    val uriHandler = LocalUriHandler.current
+
+    val onHelpClick = {
+        uriHandler.openUri(LocalSource.HELP_URL)
+    }
+
+    val onWebViewClick = f@{
+        val source = presenter.source as? HttpSource ?: return@f
+        val intent = WebViewActivity.newIntent(context, source.baseUrl, source.id, source.name)
```
Comment (arkon) [inline app/src/main/java/eu/kanade/presentation/browse/BrowseLatestScreen.kt:36]: We should avoid any overly Android-specific things from the composables so they're more reusable on other platforms. We can pass in a callback from the controller.

File: app/src/main/java/eu/kanade/presentation/browse/components/BrowseLatestToolbar.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+) {
+    AppBar(
+        navigateUp = navigateUp,
+        title = source.name,
+        actions = {
+            var selectingDisplayMode by remember { mutableStateOf(false) }
+            AppBarActions(
+                actions = listOf(
+                    AppBar.Action(
+                        title = "display_mode",
```
Comment (arkon) [inline app/src/main/java/eu/kanade/presentation/browse/components/BrowseLatestToolbar.kt:43]: Should use actual strings for these.

File: app/src/main/java/eu/kanade/presentation/browse/components/BrowseSourceToolbar.kt [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+) {
+    AppBar(
+        navigateUp = navigateUp,
+        title = source.name,
+        actions = {
+            var selectingDisplayMode by remember { mutableStateOf(false) }
+            AppBarActions(
+                actions = listOf(
+                    AppBar.Action(
+                        title = "search",
```
Comment (arkon) [inline app/src/main/java/eu/kanade/presentation/browse/components/BrowseSourceToolbar.kt:93]: Should use actual strings for these.

File: app/src/main/java/eu/kanade/presentation/browse/components/BrowseSourceToolbar.kt [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+                        }
+                    },
+                )
+            }
+        },
+    )
+}
+
+@Composable
+fun BrowseSourceSearchToolbar(
```
Comment (arkon) [inline app/src/main/java/eu/kanade/presentation/browse/components/BrowseSourceToolbar.kt:163]: Can we reuse `SearchToolbar`?

File: app/src/main/java/eu/kanade/tachiyomi/ui/browse/source/browse/BrowseSourcePresenter.kt [hunk 5]
```diff
@@ -77,55 +97,76 @@ open class BrowseSourcePresenter(
...
-     * List of filters used by the [Pager]. If empty alongside [query], the popular query is used.
-     */
-    var appliedFilters = FilterList()
+    @Composable
+    fun getManga(initialManga: DomainManga): State<DomainManga> {
+        return produceState(initialValue = initialManga, initialManga.url, initialManga.source) {
+            getManga.subscribe(initialManga.url, initialManga.source)
+                .collectLatest { manga ->
+                    if (manga == null) return@collectLatest
+                    launchIO {
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/browse/source/browse/BrowseSourcePresenter.kt:141]: Does this need to be in the global scope?

**Review by arkon — COMMENTED** [review]
Definitely just skimmed it. Some minor comments that can be addressed separately since it's a pain to track multiple revisions of big PRs on GitHub.

File: app/src/main/java/eu/kanade/tachiyomi/ui/browse/source/browse/BrowseSourcePresenter.kt (hunk 5 above)
Comment (ghostbear) [reply app/src/main/java/eu/kanade/tachiyomi/ui/browse/source/browse/BrowseSourcePresenter.kt:141]: > produceState launches a coroutine scoped to the Composition that can push values into a returned State

It's actually not in global scope

**Pull request merged**